### ISSUE TITLE
Problem building on W8/node 0.11.13

### ISSUE THREAD

**Issue by rtm** [mscdex/mmmagic#36]
My build yields errors starting with the following.

    libpcre.vcxproj -> C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\Release\\libpcre.lib
    magic.cc
    ..\src\magic.cc(18): error C2146: syntax error : missing ';' before identifier 'request' [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj]
    ..\src\magic.cc(18): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj]
    ..\src\magic.cc(39): error C2504: 'ObjectWrap' : base class undefined [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj]
    ..\src\magic.cc(61): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj]
    ..\src\magic.cc(61): error C2143: syntax error : missing ',' before '&' [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj]
    ..\src\magic.cc(109): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj]
    ..\src\magic.cc(109): error C2143: syntax error : missing ',' before '&' [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj]
    ..\src\magic.cc(147): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj]
    ..\src\magic.cc(147): error C2143: syntax error : missing ',' before '&' [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj]
    ..\src\magic.cc(193): error C2061: syntax error : identifier 'uv_work_t' [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj]
    ..\src\magic.cc(262): error C2061: syntax error : identifier 'uv_work_t' [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj]
    ..\src\magic.cc(302): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj]
    ..\src\magic.cc(302): error C2143: syntax error : missing ',' before '&' [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj]
    ..\src\magic.cc(62): error C2248: 'v8::HandleScope::HandleScope' : cannot access protected member declared in class 'v8::HandleScope' [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj]
    C:\Users\Me\.node-gyp\0.11.13\deps\v8\include\v8.h(845) : see declaration of 'v8::HandleScope::HandleScope'
    C:\Users\Me\.node-gyp\0.11.13\deps\v8\include\v8.h(829) : see declaration of 'v8::HandleScope'
    ..\src\magic.cc(71): error C2065: 'args' : undeclared identifier [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj]
    ..\src\magic.cc(71): error C2228: left of '.IsConstructCall' must have class/struct/union [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj] type is ''unknown-type''
    ..\src\magic.cc(73): error C2039: 'New' : is not a member of 'v8::String' [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj]
    C:\Users\Me\.node-gyp\0.11.13\deps\v8\include\v8.h(1661) : see declaration of 'v8::String'
    ..\src\magic.cc(73): error C2664: 'Magic::New' : cannot convert parameter 1 from 'const char [46]' to 'const int'
    [C:\cygwin64\home\me\myrepo\node_modules\mmmagic\build\magic.vcxproj] There is no context in which this conversion is possible

**Comment by mscdex** [comment]
Are you building the master branch? If not, try master. It has v0.11+ support.

**Comment by rtm** [comment]
Yes, I am building the master branch. However, I don't think this is an 0.11 issue. It's more likely a weird Windows-related build issue. I also notice something odd: when I clone the repo, it builds fine, but it refuses to build as a dependency.

**Comment by mscdex** [comment]
How are you referencing it as a dependency? How are you building it when cloning the repo?

**Comment by rtm** [comment]
# (1) Cloning and building mmmagic works fine    
    git clone ..mmmagic...
    cd mmmagic
    npm install
    # runs gyp build process and succeeds

    # (2) npm install of mmmagic fails
    mkdir newdir && cd newdir
    npm install mmmagic
    # runs gyp build process and fails with compile errors

    # (3) npm install of locally built mmmagic work fine
    npm install ../mmmagic
    # rebuilds mmmagic successfully

In other words, cloning and building mmmagic works fine. 
npm installing from github fails.
npm installing from the local repo I created succeeds.
I doubt if this is an mmmagic problem, but if anybody has any ideas it would be appreciated.

Using W8.1, cygwin, node 0.11.13, npm 1.4.9.

**Comment by mscdex** [comment]
Well `npm install mmmagic` would have been expected to fail up until a short time ago when I published the nan-ified version of mmmagic to npm :-)

Try it again and it should work for you.
